Update maliput design doc for revised RightOfWayRule API

maddog-tri opened the pull request:
And, now comes with exciting traffic diagrams!  In color!

maddog-tri commented:
+@stonier for feature-review, since he had immediate concerns about handling yield logic at intersections.  (Instructions for building the PDF are at the top of the `maliput-design.org` file.  The SVG files are Inkscape files, but there isn't necessarily any point to reviewing them directly.)

@liangfok because his work is directly affected (and because I will give him the feature-review of the code itself).

---

Review status: 0 of 7 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885#-:-LDOdkSS9MddP_tOOhn-:b-6ch2f2)*
<!-- Sent from Reviewable.io -->

stonier commented:
@maddog-tri is there a generated pdf of this anywhere?

liangfok commented:
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDluwPx6vHCtuo10awZ:-LDluwPx6vHCtuo10aw_:bop82e4) (no related file):*
BTW, when printing the document using a black-and-white laser printer (I used an HP LaserJet 1320 + Windows 10 drivers), the arrow colors are all black. Consider modifying the arrows to have other distinguishing features like dot patterns so they look OK when printed on black-and-white printers.

---

*[automotive/maliput/doc/maliput-design.org, line 579 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDldpZl1mwshQb5t4f9:-LDldpZmEVsbkCL0LwxL:b-hojzq7) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L579)):*
> ```Org
> 
> *** =RightOfWayRule=: Stopping and Yielding
> =RightOfWayRule='s describe which vehicles have right-of-way (also known as
> ```

BTW, is the apostrophe possessive? If so, consider omitting it.

---

*[automotive/maliput/doc/maliput-design.org, line 595 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlixLN5N9ZLp0FoOfT:-LDlixLN5N9ZLp0FoOfU:b-txkpd9) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L595)):*
BTW, consider alpha sorting:

> /StopAllowed/, /StopExcluded/

---

*[automotive/maliput/doc/maliput-design.org, line 605 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDliHlo8eRUVtMGZhqC:-LDliHlo8eRUVtMGZhqD:b-x3j2w6) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L605)):*
> If the type is /StopExcluded/, then vehicles are expected to stop before entering the =zone=, and to leave the =zone= if they lose the right-of-way.

I think the following qualifer should be added after "/StopExcluded/":

> and it is unsafe for the vehicle to enter the zone

Otherwise, the reader may think the vehicle must always stop before entering a zone of type "StopExcluded".

---

*[automotive/maliput/doc/maliput-design.org, line 605 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDljG7D25D_jYatUYeA:-LDljG7D25D_jYatUYeB:beal0jb) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L605)):*
> ```Org
> =zone=.  If the type is /StopExcluded/, then vehicles are expected to
> stop before entering the =zone=, and to leave the =zone= if they lose
> the right-of-way.
> ```

BTW, consider adding "as quickly as possible" after "leave the =zone=".

---

*[automotive/maliput/doc/maliput-design.org, line 614 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlkWqn1qDpP7v-jLWi:-LDlkWqn1qDpP7v-jLWj:b-47vk8z) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L614)):*
> ```Org
> | =type=     | =State::Type= enum: {/Go/, /Stop/, /StopThenGo/}              |
> |------------+---------------------------------------------------------------|
> | =yield_to= | list of =RightOfWayRule::Id=                                  |
> ```

Is there a requirement that the yield_to rules have zones that overlap with the zone of the RightOfWayRule that owns this State?

---

*[automotive/maliput/doc/maliput-design.org, line 618 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDljc4k8CE0AiN8MSLY:-LDljc4k8CE0AiN8MSLZ:byrop5r) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L618)):*
> (has right-of-way after coming to a complete stop)

This might not be sufficiently precise since a vehicle may not have right-of-way even after coming to a complete stop if another vehicle is still in the intersection.

---

*[automotive/maliput/doc/maliput-design.org, line 619 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlk2--92jZV0uDeY2-:-LDlk2--92jZV0uDeY20:b-emsvos) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L619)):*
> ```Org
> right-of-way), must /Stop/ (does not have right-of-way), or must
> /StopThenGo/ (has right-of-way after coming to a complete stop).
> The /Go/ and /StopThenGo/ types are modulated by =yield_to=, which is
> ```

BTW, I'm not sure what "modulated" means in terms of rule type. Does it mean temporarily change the rule type to be "yield", which means to stop or slow down in a manner that is safe?

---

*[automotive/maliput/doc/maliput-design.org, line 625 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDllm4G5u5OSetkb76-:-LDllm4H6BqBve_61qxV:b-8jcsx6) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L625)):*
> ```Org
> Only one =State= of a rule may be in effect at any given time.  A rule
> instance which defines only a single =State= is called a /static
> rule/; it's meaning is entirely static and fixed for all time.
> ```

/s/it's/its

---

*[automotive/maliput/doc/maliput-design.org, line 638 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlm7Du4eaznj43om_M:-LDlm7Du4eaznj43om_N:b-wlhesy) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L638)):*
> ```Org
> | =current_id=          | =State::Id=          |
> |-----------------------+----------------------|
> | =next.id=             | optional =State::Id= |
> ```

BTW, consider using "current.id" instead of "current_id" to be consistent with "next.id" and "next.duration_until".

---

*[automotive/maliput/doc/maliput-design.org, line 643 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlmHUC4xqVb_GV9D-K:-LDlmHUC4xqVb_GV9D-L:b-3mxnsw) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L643)):*
> ```Org
> |-----------------------+----------------------|
> =current_id= is the current =State= of the rule.  =next.id= is the
> /next/ =State= of the rule, if a transition is imminent and the next
> ```

Is "if a transition is imminent" a requirement? If not, consider deleting it.

I ask because I can imagine traffic light-controlled intersections where the next state of a rule is known tens of seconds beforehand, which I would not consider "imminent".

---

*[automotive/maliput/doc/maliput-design.org, line 647 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlnlhA15WdDpIV3rOh:-LDlnlhA15WdDpIV3rOi:b-jfm33g) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L647)):*
BTW, consider replacing:

> Following are discussions of

with:

> Below are

---

*[automotive/maliput/doc/maliput-design.org, line 661 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDls8GOCn3Qut-G-XDa:-LDls8GOCn3Qut-G-XDb:b-6889xu) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L661)):*
> ```Org
>     crosswalk at midblock.
>   * The pedestrian traffic route is not modeled in the road network, so only
>     one zone (for the vehicular traffic against the crosswalk) is involved.
> ```

BTW, "against" sounds odd to me. Consider "over".

---

*[automotive/maliput/doc/maliput-design.org, line 674 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDltHP7Ca-GaLo7HPGz:-LDltHP7Ca-GaLo7HPH-:b-2xfup8) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L674)):*
> from one edge of the crosswalk to the other

Consider clarifying that the `LaneSRoute` is from the southern edge to the northern edge of the crosswalk.

---

*[automotive/maliput/doc/maliput-design.org, line 677 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlteuA99cFikCOcdsc:-LDlteuA99cFikCOcdsd:b-omr4do) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L677)):*
> Note that "when it is safe to do so" is always implied with any rule.

I think this is a major point that requires highlighting earlier on in the document. It should be clarified that this predicate is intentionally vague to allow agents to interpret the rules in a manner that is appropriate for a particular region, and to reflect the fact that right-of-way rules are often vague in the real-world.

---

*[automotive/maliput/doc/maliput-design.org, line 696 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm1lQs34EFS0wJdkXJ:-LDm1lQs34EFS0wJdkXK:b-v920wb) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L696)):*
> ```Org
>   * East-west traffic is two way and uncontrolled.
>   * Northbound traffic is controlled by a stop sign.
>   * There are four zones (paths) traversing the intersection.
> ```

BTW, consider adding "as illustrated by the four arrows"

---

*[automotive/maliput/doc/maliput-design.org, line 742 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm6SXV7QVicK4AoUXF:-LDm6SXV7QVicK4AoUXG:b-y4d50i) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L742)):*
> ```Org
>   * In the signalling cycle, eastbound traffic has both a protected-left
>     (green arrow) phase and a permitted-left (flashing yellow arrow) phase.
>   * There are a total of seven zones (paths) traversing the intersection.
> ```

BTW, consider adding "as indicated by the seven path arrows" to the end of this sentence.

---

*[automotive/maliput/doc/maliput-design.org, line 770 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm7wCx0nscGW57F6P8:-LDm7wCx0nscGW57F6P9:brdu41n) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L770)):*
> ```Org
> |               | "FlashingYellow" | /Go/         | "WB/Straight", "WB/Right" |
> |---------------+------------------+--------------+---------------------------|
> | "WB/Right"    | "Red"            | /StopThenGo/ | "NB/Straight", "NB/Left"  |
> ```

I think "NB/Left" should be "EB/Left".

---

*[automotive/maliput/doc/maliput-design.org, line 777 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm78ZLC-xWqO26M64L:-LDm78ZLC-xWqO26M64M:b-9xwqzs) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L777)):*
> ```Org
> |---------------+------------------+--------------+---------------------------|
> The =State::Id='s have been chosen to loosely match the states of the
> corresponding traffic signals.
> ```

BTW, consider clarifying that it's "loose" mainly because it doesn't mention solid yellow lights. I assume the solid yellow light states are lumped into the "Green" states.

---

*[automotive/maliput/doc/maliput-design.org, line 794 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm8coOBPPmE15Q2e29:-LDm8coOBPPmE15Q2e2A:b-xbbn0j) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L794)):*
> ```Org
> to westbound traffic.  Typically in the US, the protected turn would be
> signalled by a solid green arrow, and the permitted turn by a flashing
> yellow arrow.
> ```

BTW, I'm not sure how widespread the use of a flashing yellow arrow is. In St. Louis, MO, while I was growing up, the permitted left turn is indicated by a solid green light.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[automotive/maliput/doc/maliput-design.org, line 648 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDwmAiV14D9VZh48pb1:-LDwmAiV14D9VZh48pb2:b-96i10d) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L648)):*
> ```Org
> 
> Following are discussions of possible =RightOfWayRule= configurations
> for a few example scenarios.
> ```

Can you add example scenarios that use StopAllowed?

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885)*
<!-- Sent from Reviewable.io -->

stonier commented:
*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LE549OX7AzkgnAKH4oh:-LE549OX7AzkgnAKH4oi:bfh260l) (no related file):*
Liang's already parsed this with a fine tooth comb. I'll provide some reflections (some discussed over slack) with the bigger picture in mind. 

p.12 StopAllowed, StopExcluded:  "expected to stop before entering the zone and to leave the zone if they lose the right of way" was confusing for me. Cars in an intersection or in a crosswalk, if they suddenly lose the right of way and have to suddenly yield, stop and are 'not' expected to leave the zone since they often can't or it would be dangerous to do so. Additionally, cars approaching a 'StopExcluded' crosswalk or intersection for which they have a green light are not expected to stop before entering the zone. 

This could use the crosswalk and intersection cases you described on slack to help explain & drive the definition. e.g. it seems specifically to cover the case when cars are expected to refrain from entering when, prior to entering the zone, sensed obstacles on their LaneSRoute or yield_to directions are predicted to block them from moving through without hindrance. Is this right?

Should the cases of where cars almost always break this directive be covered in the rulebook? e.g. cars turning against the flow almost always sneak into an intersection, but going straight (and obstructed by cars in-front) almost never do so, to such an extent, it's almost a rule of the road. Probably this is as simple as not setting StopExcluded on the former or having ado cars make the differentiation between LaneSRoute and yield_to obstructions. I wonder if we'll have cases where we need to pass multiple (prioritised) rules to ado cars (e.g. legal vs cultural norm). Getting off-topic now.

p.12 is StopAllowed actually a thing? I am struggling to come up with an example (and this doc doesn't have any).  Another reason to have an Example Checkbook (as noted below). Maybe should be a default? Not a blocking issue, but something to think about.

p.13 CrossWalk Example: I was partially thrown by this before appreciating the StopExcluded properly. However, there remains the fact that this right of way rule doesn't provide any hint to an ado car that it should treat the situation of having a pedestrian waiting on the side of the crosswalk (looking out onto the road) vs a pedestrian waiting on the side of the road anywhere else. As such, it doesn't appear to be a fully functional RightOfWay rule for crosswalks (that ado cars can use). Matt, you mentioned that you'd thought that this should perhaps be handled by 'pedestrian' roads with yield_to marking them. Should mention that in at least a note here.

RoadRule Example CheckBook: Would a longer list of examples alongside their rules for the purposes of checking completeness and providing a reference for lookup be useful? This may be overkill, but I'd be far more comfortable if at least one person has run this Gedankenexperiment and can enumerate the examples they've worked through so we can accept a reasonable completeness argument on something other than blind faith.

Not relevant to the road rulebook, however:

all: I wonder if this doc would be better served by not conflating (occasionally) conceptual design with computational design considerations?

p.5 on driveable vs modelled volumes:  This could use a diagram to quickly grok the problem rather than going via text. 

Questions we can take offline:

p.5 Question on H_{driveable}: I can imagine features detected by radar simply being 'connected' to the road surface via (s,r,h) coordinates without having to be bounded inside a volume as well. Are there other use cases in mind that motivate the driveable height as a bounded volume concept? It's also discussed further down as having to maintain G^1 continuity along lines of constant h, however G^1 continuity has hitherto only been mentioned as a requirement of the 2-dimensional continuous map from \mathbb{R}^2 \rightarrow \mathbb{R}^3.

p.5 Question on driveable vs modelled volumes: What are the consequences of cases in which, as noted, there can be a discrepancy? Are ado cars expected to drive on the driveable or modelled surface? Is the fact that the G1 constraints, as formulated, cannot represent certain driveable surfaces a problem (I am imagining a corner on a flat world with centreline that has a radius of curvature < width of the road)?

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885)*
<!-- Sent from Reviewable.io -->

maddog-tri commented:
*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LE549OX7AzkgnAKH4oh:-LE7baqn194hlI7vG2TN:b-cvv9mi) (no related file):*
<details><summary><i>Previously, stonier (Daniel Stonier) wrote…</i></summary><blockquote>

Liang's already parsed this with a fine tooth comb. I'll provide some reflections (some discussed over slack) with the bigger picture in mind. 

p.12 StopAllowed, StopExcluded:  "expected to stop before entering the zone and to leave the zone if they lose the right of way" was confusing for me. Cars in an intersection or in a crosswalk, if they suddenly lose the right of way and have to suddenly yield, stop and are 'not' expected to leave the zone since they often can't or it would be dangerous to do so. Additionally, cars approaching a 'StopExcluded' crosswalk or intersection for which they have a green light are not expected to stop before entering the zone. 

This could use the crosswalk and intersection cases you described on slack to help explain & drive the definition. e.g. it seems specifically to cover the case when cars are expected to refrain from entering when, prior to entering the zone, sensed obstacles on their LaneSRoute or yield_to directions are predicted to block them from moving through without hindrance. Is this right?

Should the cases of where cars almost always break this directive be covered in the rulebook? e.g. cars turning against the flow almost always sneak into an intersection, but going straight (and obstructed by cars in-front) almost never do so, to such an extent, it's almost a rule of the road. Probably this is as simple as not setting StopExcluded on the former or having ado cars make the differentiation between LaneSRoute and yield_to obstructions. I wonder if we'll have cases where we need to pass multiple (prioritised) rules to ado cars (e.g. legal vs cultural norm). Getting off-topic now.

p.12 is StopAllowed actually a thing? I am struggling to come up with an example (and this doc doesn't have any).  Another reason to have an Example Checkbook (as noted below). Maybe should be a default? Not a blocking issue, but something to think about.

p.13 CrossWalk Example: I was partially thrown by this before appreciating the StopExcluded properly. However, there remains the fact that this right of way rule doesn't provide any hint to an ado car that it should treat the situation of having a pedestrian waiting on the side of the crosswalk (looking out onto the road) vs a pedestrian waiting on the side of the road anywhere else. As such, it doesn't appear to be a fully functional RightOfWay rule for crosswalks (that ado cars can use). Matt, you mentioned that you'd thought that this should perhaps be handled by 'pedestrian' roads with yield_to marking them. Should mention that in at least a note here.

RoadRule Example CheckBook: Would a longer list of examples alongside their rules for the purposes of checking completeness and providing a reference for lookup be useful? This may be overkill, but I'd be far more comfortable if at least one person has run this Gedankenexperiment and can enumerate the examples they've worked through so we can accept a reasonable completeness argument on something other than blind faith.

Not relevant to the road rulebook, however:

all: I wonder if this doc would be better served by not conflating (occasionally) conceptual design with computational design considerations?

p.5 on driveable vs modelled volumes:  This could use a diagram to quickly grok the problem rather than going via text. 

Questions we can take offline:

p.5 Question on H_{driveable}: I can imagine features detected by radar simply being 'connected' to the road surface via (s,r,h) coordinates without having to be bounded inside a volume as well. Are there other use cases in mind that motivate the driveable height as a bounded volume concept? It's also discussed further down as having to maintain G^1 continuity along lines of constant h, however G^1 continuity has hitherto only been mentioned as a requirement of the 2-dimensional continuous map from \mathbb{R}^2 \rightarrow \mathbb{R}^3.

p.5 Question on driveable vs modelled volumes: What are the consequences of cases in which, as noted, there can be a discrepancy? Are ado cars expected to drive on the driveable or modelled surface? Is the fact that the G1 constraints, as formulated, cannot represent certain driveable surfaces a problem (I am imagining a corner on a flat world with centreline that has a radius of curvature < width of the road)?

</blockquote></details>

Thanks, and welcome to Reviewable.  Could you copy these comments to the appropriate places in the `maliput-design.org` source below?  (Or, at least split them into multiple top-level threads?)  It's hard to respond to and converse about the variety of issues when they are all thrown together into a single block of text.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885)*
<!-- Sent from Reviewable.io -->

stonier commented:
*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LE549OX7AzkgnAKH4oh:-LEQ5wgg2YTZZT01oYnv:b-64gkj2) (no related file):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

Thanks, and welcome to Reviewable.  Could you copy these comments to the appropriate places in the `maliput-design.org` source below?  (Or, at least split them into multiple top-level threads?)  It's hard to respond to and converse about the variety of issues when they are all thrown together into a single block of text.
</blockquote></details>

Please just spin up new, separate discussion threads with your responses if that is the way you'd like to move it forward.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885)*
<!-- Sent from Reviewable.io -->

maddog-tri commented:
Alright, I think I have addressed all the comments.  (I split up Daniel's unReviewable blob into multiple comments by topic/place, but that means those new comments were all initiated by me.)

-@stonier and +@liangfok for feature review, since Liang had the bulk of the comments anyway.

---

Review status: 15 unresolved discussions, LGTM missing from assignee stonier, platform LGTM missing

---

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LE549OX7AzkgnAKH4oh:-LEpwixZ7VB7aXsR9oEL:b-896fix) (no related file):*
<details><summary><i>Previously, stonier (Daniel Stonier) wrote…</i></summary><blockquote>

Please just spin up new, separate discussion threads with your responses if that is the way you'd like to move it forward.
</blockquote></details>

Done.

---

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LEpsNVE1gm_3TLBQYDv:-LEpsNVF6Ch1QVnjwMVz:biozkm4) (no related file):*
> p.5 Question on H_{driveable}: I can imagine features detected by radar simply being 'connected' to the road surface via (s,r,h) coordinates without having to be bounded inside a volume as well. Are there other use cases in mind that motivate the driveable height as a bounded volume concept? It's also discussed further down as having to maintain G^1 continuity along lines of constant h, however G^1 continuity has hitherto only been mentioned as a requirement of the 2-dimensional continuous map from \mathbb{R}^2 \rightarrow \mathbb{R}^3.

Yes, the value of the "bounded volume" concept becomes apparent in situations like this, where expressing the network as a tangle of tubes makes it easier to understand how it partitions the world's 3-space:
![Spaghetti Junction](https://upload.wikimedia.org/wikipedia/commons/b/bd/Spaghetti-Junction-Crop.jpg)
By Original - <a rel="nofollow" class="external text" href="https://www.flickr.com/photos/highwaysagency/4970763078/">Highways Agency photo on flickr</a> - Own work derived from <a href="//commons.wikimedia.org/wiki/File:Aerial_view_of_M6A38_Spaghetti_Junction.jpg" title="File:Aerial view of M6A38 Spaghetti Junction.jpg">File:Aerial view of M6A38 Spaghetti Junction.jpg</a>, whose license permits such derivative work when properly credited, <a href="https://creativecommons.org/licenses/by/2.0" title="Creative Commons Attribution 2.0">CC BY 2.0</a>, <a href="https://commons.wikimedia.org/w/index.php?curid=19927460">Link</a>

---

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LEpsPZ64ZEOUh4HEp2r:-LEpsPZ64ZEOUh4HEp2s:b-ll4afd) (no related file):*
> p.5 Question on driveable vs modelled volumes: What are the consequences of cases in which, as noted, there can be a discrepancy? Are ado cars expected to drive on the driveable or modelled surface? Is the fact that the G1 constraints, as formulated, cannot represent certain driveable surfaces a problem (I am imagining a corner on a flat world with centreline that has a radius of curvature < width of the road)?

Clarifying the terminology: the "driveable" volume in maliput's LANE-space is always a subset of the "modelled" volume (because the model cannot describe anything outside of the model).  The "modelled" volume could be a subset of the actual "physical" volume, and there could indeed be asphalt beyond the modelled volume. Ado cars driving in a lane are expected to drive on the modelled surface of that lane.

 The reason for the "modelled" bounds is to avoid singularities in the World->Lane map; it's unavoidable if one parameterizes the world in terms of curves parallel to a nominal centerline path.  I don't think this is a problem for the lane-based traffic that we are focused on. When more free-form driving becomes an issue (e.g., doing donuts in a wide open parking lot), we may need to have ado cars that can break free of the lane parameterization (but even in those cases, there will likely be lanes that traverse the free-form area, and many vehicles parameterizing the world in terms of those lanes).

---

*[automotive/maliput/doc/maliput-design.org, line 579 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDldpZl1mwshQb5t4f9:-LEu8Am0BuUEXEKwopH8:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L579)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, is the apostrophe possessive? If so, consider omitting it.
</blockquote></details>

Done.

---

*[automotive/maliput/doc/maliput-design.org, line 605 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LEpqxkCBHHvnwsPxrn_:-LEpqxkCBHHvnwsPxrna:b-sf1c4s) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L605)):*
Daniel's comment:
> p.12 StopAllowed, StopExcluded: "expected to stop before entering the zone and to leave the zone if they lose the right of way" was confusing for me. Cars in an intersection or in a crosswalk, if they suddenly lose the right of way and have to suddenly yield, stop and are 'not' expected to leave the zone since they often can't or it would be dangerous to do so. Additionally, cars approaching a 'StopExcluded' crosswalk or intersection for which they have a green light are not expected to stop before entering the zone.

> This could use the crosswalk and intersection cases you described on slack to help explain & drive the definition. e.g. it seems specifically to cover the case when cars are expected to refrain from entering when, prior to entering the zone, sensed obstacles on their LaneSRoute or yield_to directions are predicted to block them from moving through without hindrance. Is this right?

> Should the cases of where cars almost always break this directive be covered in the rulebook? e.g. cars turning against the flow almost always sneak into an intersection, but going straight (and obstructed by cars in-front) almost never do so, to such an extent, it's almost a rule of the road. Probably this is as simple as not setting StopExcluded on the former or having ado cars make the differentiation between LaneSRoute and yield_to obstructions. I wonder if we'll have cases where we need to pass multiple (prioritised) rules to ado cars (e.g. legal vs cultural norm). Getting off-topic now.

I changed the language to hopefully clarify.

To the last paragraph:  yeah, the primary goal here is to express the rules of the road.  If we need agents to tactically break/stretch the rules, that probably belongs somewhere else.  ("Sneaking into an intersection" isn't _technically_ breaking the _StopExcluded_ rule as long as there is room ahead to exit the intersection (after yielding to oncoming traffic, which is expected to stop coming while the light is yellow).)

---

*[automotive/maliput/doc/maliput-design.org, line 605 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LEprKOn1XGTOKNeK1xe:-LEprKOn1XGTOKNeK1xf:b-4b4lgw) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L605)):*
BTW, Daniel's comment:
> p.12 is StopAllowed actually a thing? I am struggling to come up with an example (and this doc doesn't have any). Another reason to have an Example Checkbook (as noted below). Maybe should be a default? Not a blocking issue, but something to think about.

I've added an example with StopAllowed (as also requested by Liang).

---

*[automotive/maliput/doc/maliput-design.org, line 605 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDliHlo8eRUVtMGZhqC:-LEw8CJu6DAaWO5MWrvZ:b-ts810c) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L605)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

> If the type is /StopExcluded/, then vehicles are expected to stop before entering the =zone=, and to leave the =zone= if they lose the right-of-way.

I think the following qualifer should be added after "/StopExcluded/":

> and it is unsafe for the vehicle to enter the zone

Otherwise, the reader may think the vehicle must always stop before entering a zone of type "StopExcluded".
</blockquote></details>

I changed the language to hopefully clarify.  (The point here isn't safety (which always trumps everything), but rather not getting stuck in an intersection and blocking traffic.)

---

*[automotive/maliput/doc/maliput-design.org, line 614 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlkWqn1qDpP7v-jLWi:-LDrTJ6BAAMtVajSJtDU:b-4jo5b2) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L614)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Is there a requirement that the yield_to rules have zones that overlap with the zone of the RightOfWayRule that owns this State?
</blockquote></details>

There is no such requirement.  One could put non-overlapping rules in a yield_to list, but they wouldn't have any effect, because you can't yield to another vehicle if it isn't actually possible to get in its way.

---

*[automotive/maliput/doc/maliput-design.org, line 618 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDljc4k8CE0AiN8MSLY:-LEuCy1TElWFa5u6udrV:bhumfqn) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L618)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

> (has right-of-way after coming to a complete stop)

This might not be sufficiently precise since a vehicle may not have right-of-way even after coming to a complete stop if another vehicle is still in the intersection.
</blockquote></details>

I think this is now covered by the '"right of way" does not mean "right to smash through obstacles"' note at the beginning of the RightOfWayRule section.

---

*[automotive/maliput/doc/maliput-design.org, line 619 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlk2--92jZV0uDeY2-:-LEuDSIvE12mYYzdCrSm:bg1wq1k) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L619)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, I'm not sure what "modulated" means in terms of rule type. Does it mean temporarily change the rule type to be "yield", which means to stop or slow down in a manner that is safe?
</blockquote></details>

I added a sentence to clarify; let me know if that works.

---

*[automotive/maliput/doc/maliput-design.org, line 625 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDllm4G5u5OSetkb76-:-LEuDhwo6TnM2OYanafv:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L625)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

/s/it's/its
</blockquote></details>

Done.

---

*[automotive/maliput/doc/maliput-design.org, line 638 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlm7Du4eaznj43om_M:-LEuDkrwAUGF47AjhMoU:b-8syyco) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L638)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider using "current.id" instead of "current_id" to be consistent with "next.id" and "next.duration_until".
</blockquote></details>

The C++ expressions are in fact `current_id` and `next.id` because of how the (nested) struct is laid out.

---

*[automotive/maliput/doc/maliput-design.org, line 643 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlmHUC4xqVb_GV9D-K:-LEuEfy_5dLiiDV-3gZD:bahn369) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L643)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Is "if a transition is imminent" a requirement? If not, consider deleting it.

I ask because I can imagine traffic light-controlled intersections where the next state of a rule is known tens of seconds beforehand, which I would not consider "imminent".
</blockquote></details>

Done. (by matching to language in the corresponding header docstring, which had had the same problem.)

---

*[automotive/maliput/doc/maliput-design.org, line 648 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LEprur60ine82XQmgX9:-LEprur60ine82XQmgXA:bkmkokh) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L648)):*
Daniel's comment:
> RoadRule Example CheckBook: Would a longer list of examples alongside their rules for the purposes of checking completeness and providing a reference for lookup be useful? This may be overkill, but I'd be far more comfortable if at least one person has run this Gedankenexperiment and can enumerate the examples they've worked through so we can accept a reasonable completeness argument on something other than blind faith.

I definitely _feel_ like I have given this design a lot of thought, but I have not enumerated every single scenario that has crossed my mind.  If anyone else wants to ponder scenarios, too, that would be great.

I can add more examples (in a follow-up PR, preferably), but someone else is going to have to brainstorm a list of additional scenarios that are of concern, since I already documented the examples that I thought were the most critical to exercising the design.  (Keep in mind that documenting examples is not free; it takes some time to draw good diagrams and write good explanations.)

---

*[automotive/maliput/doc/maliput-design.org, line 648 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDwmAiV14D9VZh48pb1:-LEwZfKy7Bh_u7gO9yki:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L648)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Can you add example scenarios that use StopAllowed?
</blockquote></details>

Done.

---

*[automotive/maliput/doc/maliput-design.org, line 661 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDls8GOCn3Qut-G-XDa:-LEuFe9LDwQ1REejHsg5:br70vo0) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L661)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, "against" sounds odd to me. Consider "over".
</blockquote></details>

Done (something).

---

*[automotive/maliput/doc/maliput-design.org, line 674 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDltHP7Ca-GaLo7HPGz:-LEuG3nnD41gzbEXS2-D:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L674)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

> from one edge of the crosswalk to the other

Consider clarifying that the `LaneSRoute` is from the southern edge to the northern edge of the crosswalk.
</blockquote></details>

Done.

---

*[automotive/maliput/doc/maliput-design.org, line 677 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDlteuA99cFikCOcdsc:-LEwL3T31A9-YbdEfzV9:brliu4p) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L677)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

> Note that "when it is safe to do so" is always implied with any rule.

I think this is a major point that requires highlighting earlier on in the document. It should be clarified that this predicate is intentionally vague to allow agents to interpret the rules in a manner that is appropriate for a particular region, and to reflect the fact that right-of-way rules are often vague in the real-world.
</blockquote></details>

I've added a big footnote at the beginning of this section which I think addresses this.  The issue isn't that the rules of the road are vague; it's that the rules of the road always come second to the rules of physics.

---

*[automotive/maliput/doc/maliput-design.org, line 684 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LEpr_KQEIV8CHbHc3LF:-LEpr_KQEIV8CHbHc3LG:bmzwudl) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L684)):*
Daniel's comment:
> p.13 CrossWalk Example: I was partially thrown by this before appreciating the StopExcluded properly. However, there remains the fact that this right of way rule doesn't provide any hint to an ado car that it should treat the situation of having a pedestrian waiting on the side of the crosswalk (looking out onto the road) vs a pedestrian waiting on the side of the road anywhere else. As such, it doesn't appear to be a fully functional RightOfWay rule for crosswalks (that ado cars can use). Matt, you mentioned that you'd thought that this should perhaps be handled by 'pedestrian' roads with yield_to marking them. Should mention that in at least a note here.

Done --- note added.

---

*[automotive/maliput/doc/maliput-design.org, line 696 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm1lQs34EFS0wJdkXJ:-LEuGYRgEUEPu6ZSM2Hf:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L696)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider adding "as illustrated by the four arrows"
</blockquote></details>

Done.

---

*[automotive/maliput/doc/maliput-design.org, line 742 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm6SXV7QVicK4AoUXF:-LEuGfNv8rS-czEm5Lxh:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L742)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider adding "as indicated by the seven path arrows" to the end of this sentence.
</blockquote></details>

Done.

---

*[automotive/maliput/doc/maliput-design.org, line 770 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm7wCx0nscGW57F6P8:-LEuGuvf0ulge4ZDCM_Z:b-lztb15) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L770)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

I think "NB/Left" should be "EB/Left".
</blockquote></details>

Done. (Good catch.  Originally I drew this scenario with the two-way traffic going N-S, but then I rotated it to better fit the page, and had to fix up the table accordingly.)

---

*[automotive/maliput/doc/maliput-design.org, line 777 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm78ZLC-xWqO26M64L:-LEuJ9Hc0KkwPUUD85s1:bcykji0) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L777)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider clarifying that it's "loose" mainly because it doesn't mention solid yellow lights. I assume the solid yellow light states are lumped into the "Green" states.
</blockquote></details>

Done (I think), and your assumption is correct.

---

*[automotive/maliput/doc/maliput-design.org, line 794 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LDm8coOBPPmE15Q2e29:-LEuwRI6Aoy02YGVD-bZ:byahcwr) ([raw file](https://github.com/robotlocomotion/drake/blob/a69d70526e29cfd5c0b62cd19a1106d447521b7c/automotive/maliput/doc/maliput-design.org#L794)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, I'm not sure how widespread the use of a flashing yellow arrow is. In St. Louis, MO, while I was growing up, the permitted left turn is indicated by a solid green light.
</blockquote></details>

Yeah, I was getting a little sloppy there.  Portland has a lot of flashing-yellow permitted-left-turns, but I don't think I had ever seen them before.  I think it may be a relatively new (but definitely improved) addition to the MUTCD.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885#-:-LEz9RIF7Tzos4RPIagD:btai7fj)*
<!-- Sent from Reviewable.io -->

liangfok commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r2.
Review status: 7 unresolved discussions, platform LGTM missing

---

*[automotive/maliput/doc/maliput-design.org, line 798 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LF39tBO11tJf6YwBGEm:-LF39tBO11tJf6YwBGEn:b-wlxq9l) ([raw file](https://github.com/robotlocomotion/drake/blob/0aad9018d8aef4838abbe93bb3b3129be177682d/automotive/maliput/doc/maliput-design.org#L798)):*
> ```Org
> |---------------+------------------+--------------+---------------------------|
> The =State::Id='s have been chosen to loosely match the states of the
> corresponding traffic signals.  (Note that a typically a "yellow light"
> ```

BTW, extra "a" before "typically".

---

*[automotive/maliput/doc/maliput-design.org, line 828 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LF3ADI83S-XP22YqZ1a:-LF3ADI92Jjko-fn92s3:b-61fd4k) ([raw file](https://github.com/robotlocomotion/drake/blob/0aad9018d8aef4838abbe93bb3b3129be177682d/automotive/maliput/doc/maliput-design.org#L828)):*
> ```Org
> Figure [[fig:RoWR-freeway-merge]] is a scenario with a freeway merge:
>   * Freeway has two lanes of eastbound traffic.
>   * Entrance ramp merges onto to freeway from the right (south).
> ```

/s/to/the/

---

*[automotive/maliput/doc/maliput-design.org, line 849 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LF3Ak792WT4w59X0FsY:-LF3Ak792WT4w59X0FsZ:bt2kiuw) ([raw file](https://github.com/robotlocomotion/drake/blob/0aad9018d8aef4838abbe93bb3b3129be177682d/automotive/maliput/doc/maliput-design.org#L849)):*
> ```Org
> /StopAllowed/.  That means there are no "stop lines" (real or
> implicit) and no exclusion zones that are expected to be left
> unblocked by stopped traffic.  Both rules' static states are of type
> ```

BTW, consider explicitly mentioning that the Freeway uses a StopAllowed rule even though many freeways have minimum speed limits. The implication here is that vehicles can stop in the zone, but should not stop if they don't have to.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885#-:-LF3BKm--q_yNg3rdrx-:bnfp4nl)*
<!-- Sent from Reviewable.io -->

maddog-tri commented:
+@sherm1 for platform review, please.

---

Review status: 7 unresolved discussions, platform LGTM missing

---

*[automotive/maliput/doc/maliput-design.org, line 828 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8885#-LF3ADI83S-XP22YqZ1a:-LF3KjmMEimoZRXzpPeC:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/0aad9018d8aef4838abbe93bb3b3129be177682d/automotive/maliput/doc/maliput-design.org#L828)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

/s/to/the/
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885#-:-LF3LD0TEzqszbMdQCH8:bppyoqq)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
@maddog-tri: I see a ".org" file here and a bunch of binary files. Not clear what I'm supposed to do with them? Is there a way to build the document to look at it formatted?

---

Review status: 7 unresolved discussions, LGTM missing from assignee sherm1, platform LGTM missing

---



*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885#-:-LF3jggy-sHBa9jkwfkW:b-iph1ay)*
<!-- Sent from Reviewable.io -->

liangfok commented:
There's a comment at the top of the .org file that explains how to generate the PDF. It requires emacs.

---

Reviewed 1 of 1 files at r3.
Review status: 6 unresolved discussions, LGTM missing from assignee sherm1, platform LGTM missing

---



*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885#-:-LF3qfMe5_o0ZrtR9C2-:bdtkut2)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Thanks, @liangfok -- I followed the instructions and got a nice pdf!

---

Review status: 6 unresolved discussions, LGTM missing from assignee sherm1, platform LGTM missing

---



*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885#-:-LF4d5dTCM6WT4AMneD8:b-2mivcw)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Beautiful document! I'm assuming the contents have been reviewed and didn't attempt that. The instructions are easy to follow and produce a pdf with all the figures and tables looking good, so platform <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>.

---

Review status: 6 unresolved discussions, platform LGTM from [sherm1]

---



*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8885#-:-LF4fDM36dCNOKG5aXLU:b-sir9pp)*
<!-- Sent from Reviewable.io -->
